File: api/src/endpoint/repository/delete_repository.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct DeleteRepository;

impl Endpoint for DeleteRepository {
    const PATH: &'static str = "/repository/{owner}/{repo}";
    const METHOD: http::Method = http::Method::DELETE;

    type Request = DeleteRepositoryRequest;
    type Response = DeleteRepositoryResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct DeleteRepositoryRequest {}

pub type DeleteRepositoryResponse = ();

File: gitdot-api-ts/src/endpoint/repository/delete-repository.ts
import { z } from "zod";

export const DeleteRepositoryRequest = z.object({});
export type DeleteRepositoryRequest = z.infer<typeof DeleteRepositoryRequest>;

export const DeleteRepository = {
  path: "/repository/{owner}/{repo}",
  method: "DELETE",
  request: DeleteRepositoryRequest,
  response: z.void(),
} as const;
export type DeleteRepository = typeof DeleteRepository;
